### DISCUSSION TITLE
Defining Field for array of string values prop

### DISCUSSION THREAD

**Discussion by sagarchoudhary96** [puckeditor/puck#68] [Q&A]
My component have a prop tags of array type but values as string for it ``tags: string[]` . how can i define Field config for this prop, seems like it's currently not supported.

**Comment by Danm72** [comment]
If i'm understanding you correctly you can add this to your fields config:
```typescript
tags: {
            type: "array",
            getItemSummary: (item) => item.tag || "Tag",
            arrayFields: {
                tag: {type: "text"},
            },
        },
```

**Comment by chrisvxd** [comment]
Correct, we only support objects in `array` fields at the time-being, so the easiest thing to do is as @Danm72 proposed.

If your component expects `string[]`, suggest you could wrap your component in another one to map from the object to string.

**Reply by sagarchoudhary96** [reply]
adding a wrapper could be tough in case of having many components and having props as string[] at any nested level in the props.

**Reply by chrisvxd** [reply]
That's interesting. What sort of UI would you expect here?

**Reply by sagarchoudhary96** [reply]
It could be simple Ui same as current array ui which shows textfield on new item button click where we can enter the string value for the item and in field props can have prop as:

```

{
  "type": "array",
  "arrayType": "simple" | "object" --> something similar to mark that its just array of values instead of arrayFields
}
```

**Reply by chrisvxd** [reply]
I've been thinking about this.

How long are the strings you're defining? I wonder if we can leverage a tag / multi-select style interface for this, instead of reusing the `array` type.

Similar to the [multi](https://react-select.com/home) configuration of react-select, with optional auto complete items:

<img width="585" alt="image" src="https://github.com/measuredco/puck/assets/985961/73866f42-0fca-4bbf-9f0f-45562536deee">

**Reply by sagarchoudhary96** [reply]
@chrisvxd do we have any updates on this, i.e to be able to define array as strings array instead of object.

**Reply by chrisvxd** [reply]
@sagarchoudhary96 I think you could implement a custom field to achieve this using react-select. Let me know if that works.
